### DISCUSSION TITLE
What necessitates using 3.8?

### DISCUSSION THREAD

**Discussion by PySimpleGUI** [pls-rs/pls#21] [Help Q&A] [answered]
What in 3.8 are you using that's not available in 3.6?  Have you considered trying to cover as many Python versions as possible?

**Comment by dhruvkb** [comment, marked as answer]
Python 3.6 has [reached end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches), so there are no plans to support it at all. In all honesty, I went with Python 3.8 as the minimum supported version because that's the first version supported natively on my M1 MacBook development machine.

While expanding support for more Python versions is definitely worth the effort, I'm the only developer actively working on the project and I just can't seem to get Python 3.7 to work locally. I'm happy to accept PRs that can make the project backwards compatible to Python 3.7.

**Reply by PySimpleGUI** [reply]
Thanks much for the reply and thoughtful answer!  I'm always curious about Python versions projects support or don't to see if there is a critical feature that I may or may not be missing.
f-strings have been the most difficult to live without in my core code.  I'm 3.6 compatible in my demo programs so I at least have them there.

I didn't know that 3.7 didn't run on the M1.  The Mac is so tricky with Python and with tkinter in particular.
